R/functions.R

```diff
@@ -49,26 +61,72 @@ collapse_pm_query <- function(df) {
   entry <- dplyr::group_by(df, pmid, doi, title, abstract, year, journal, entity_name)
   entry <- summarise(entry, authors = toString(fullname))
   if (nchar(entry$authors) > 500) {
-    entry$authors <- str_trunc(entry$authors, width = 500)
+    entry$authors <- stringr::str_trunc(entry$authors, width = 500)
   }
   entry
 }
-#' Wrapper for Pubmed Query
+#' Wrapper for Pubmed Query using Pubmed IDs
 #' This function wraps several functions:
 #' - query Pubmed
 #' - create an entity name from first author, journal, year and Id
 #' - abbreviates the author names by first initial, last name
 #' - creates one row per PubmedId
 #' @param vec A character vector of PudmedIds.
+#' @param sleep_time Amount of time, in seconds, to sleep between queries.
+#' Without APIKey, Pubmed limits rate of queries to 3 per second. Default of
+#' 0.35 seconds.
 #' @export
-query_list_pmids <- function(vec) {
-  vec <- purrr::map(vec, ~ query_pm(.))
+query_list_pmids <- function(vec, sleep_time = 0.35) {
+  vec <- purrr::map(vec, ~ query_pm(., sleep_time = sleep_time))
   vec <- purrr::map(vec, ~ make_entity_name(.))
   vec <- purrr::map(vec, ~ collapse_names(.))
   vec <- purrr::map(vec, ~ collapse_pm_query(.))
   df <- data.frame(Reduce(rbind, vec))
   df
 }
+#' Wrapper for Pubmed Query using general queries
+#' This function wraps several functions:
+#' - query Pubmed
+#' - create an entity name from first author, journal, year and Id
+#' - abbreviates the author names by first initial, last name
+#' - creates one row per PubmedId
+#' - query without associated PubmedIds will be left out of returned data
+#' - query added as an additional column
+#' @param queries A character vector of queries.
+#' @param sleep_time Amount of time, in seconds, to sleep between queries.
+#' Without APIKey, Pubmed limits rate of queries to 3 per second. Default of
+#' 0.35 seconds.
+#' @export
+query_list_general <- function(queries, sleep_time = 0.35) {
+  vec <- purrr::map(queries, ~ query_general(., sleep_time = sleep_time))
+  # Limit queries to only those associated with pubmedIDs
+  queries <- queries[!is.na(vec)]
+  vec <- vec[!is.na(vec)]
+  dfs <- purrr::map(vec, ~ query_list_pmids(.))
+  dfs <- purrr::map2(dfs, queries, function (df, query) {
+    tibble::add_column(df, query = query)
+  })
+  df <- Reduce(rbind, dfs)
+  df
+}
+#' Query for Pubmed IDs only
+#'
+#' @param query Character string with query
```

Comment:
```suggestion
#' @param query Character string with query
```
Could you provide an example query?

Reply:
Sure. I'll add that.